driver_test.go

```diff
@@ -12,26 +12,272 @@
+			option.WithEndpoint(spannerHost),
+			option.WithGRPCDialOption(grpc.WithInsecure()))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		adminClient, err = adminapi.NewDatabaseAdminClient(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return adminClient, nil
+}
+
+func (c *Connector) Close() {
+	c.client.Close()
+	c.adminClient.Close()
+}
+
+func init() {
+
+	var projectId, instanceId, databaseId string
+	var ok bool
+
+	// Get environment variables or set to default.
+	if instanceId, ok = os.LookupEnv("SPANNER_TEST_INSTANCE"); !ok {
+		instanceId = "test-instance"
+	}
+	if projectId, ok = os.LookupEnv("SPANNER_TEST_PROJECT"); !ok {
+		projectId = "test-project"
+	}
+	if databaseId, ok = os.LookupEnv("SPANNER_TEST_DBID"); !ok {
+		databaseId = "gotest"
+	}
+
+	// Derive data source name.
+	dsn = "projects/" + projectId + "/instances/" + instanceId + "/databases/" + databaseId
+}
+
+// Executes DDL statements.
+func executeDdlApi(conn *Connector, ddls []string) error {
+
+	op, err := conn.adminClient.UpdateDatabaseDdl(conn.ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database:   dsn,
+		Statements: ddls,
+	})
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(conn.ctx); err != nil {
+		return err
 	}
-	connector, err := driver.OpenConnector("projects/$PROJECT/instances/$INSTANCE/databases/$DATABASE")
+	return nil
+}
+
+// Executes DML using the client library.
+func ExecuteDMLClientLib(dml []string) error {
+
+	// Open client/
+	ctx := context.Background()
+	client, err := spanner.NewClient(ctx, dsn)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	// Put strings into spanner.Statement structure.
+	var stmts []spanner.Statement
+	for _, line := range dml {
+		stmts = append(stmts, spanner.NewStatement(line))
+	}
+
+	// Execute statements.
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		_, err := txn.BatchUpdate(ctx, stmts)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	return err
+}
+
+func TestQueryContext(t *testing.T) {
+
+	// Set up test table.
+	conn, err := NewConnector()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	err = executeDdlApi(conn, []string{`CREATE TABLE TestQueryContext (
+		A   STRING(1024),
+		B  STRING(1024),
+		C   STRING(1024)
+	)	 PRIMARY KEY (A)`})
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ExecuteDMLClientLib([]string{`INSERT INTO TestQueryContext (A, B, C) 
+		VALUES ("a1", "b1", "c1"), ("a2", "b2", "c2") , ("a3", "b3", "c3") `})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Open db.
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	type testQueryContextRow struct {
+		A, B, C string
+	}
+
+	tests := []struct {
+		name           string
+		input          string
+		want           []testQueryContextRow
+		wantErrorQuery bool
+		wantErrorScan  bool
+		wantErrorClose bool
+	}{
+		{name: "empty query",
```

Comment:
Each field should be on a new line include the first one i.e. 
```
{
    name: "empty query",
    wantErrorClose: true,
    input: "",
    want: []testQueryContextRow{},
}
```
`go fmt` will then make the values line up which is more readable :+1:

Reply:
fixed